src/components/Calendar.vue

```diff
@@ -0,0 +1,147 @@
+<template>
+  <div id="calendarFull">
+    <button @click="refreshEvents">Refresh</button>
+    <button v-if="selected._id" @click="removeEvent">Remove</button>
+    <pre v-if="selected._id">Title: {{selected.title}}
+    Start: {{selected.start}}
+    End: {{selected.end}}
+    Allday: {{selected.allDay}}
+    ID: {{selected.id}}</pre>
+    <full-calendar ref="calendar" :events="events" @event-drop="eventDrop" @event-resize="eventResize" @event-selected="eventSelected" @event-created="eventCreated" :config="config"></full-calendar>
+  </div>
+</template>
+
+<script>
+import frappe from "frappejs";
+import Vue from 'vue'
+import FullCalendar from 'vue-full-calendar';
+const { DateTime } = require('luxon');
+Vue.use(FullCalendar);
+
+const getCircularReplacer = () => {
```

Comment:
If this is not specific to this component, move this to a utils file?

Reply:
As for now, I have only used it here, and haven't seen it being used anywhere else. It's basically used to solve cyclic references for objects.